work_queue/src/work_queue.c

```diff
@@ -3965,6 +3972,57 @@ static struct work_queue_worker *find_worker_by_time(struct work_queue *q, struc
 	}
 }
 
+
+// checks workers against tasks and returns if a task has the  
+// possibility of fitting a worker
+static int check_task_fit_in_worker(struct work_queue *q, struct work_queue_task *t, struct work_queue_worker *w)
+{
+	int set = 0;
+	struct rmsummary *l = task_worker_box_size(q,w,t);
+
+	// baseline resurce comparison of worker total resources and a task requested resorces
+
+	if((double)w->resources->cores.total < l->cores ) {
+		set = set | (1<<0);
+	}
+
+	if((double)w->resources->memory.total < l->memory ) {
+		set = set | (1<<1);
+	}
+
+	if((double)w->resources->disk.total < l->disk ) { 
+		set = set | (1<<2);
+	}
+
+	if((double)w->resources->gpus.total < l->gpus ) {
+		set = set | (1<<3);
+	}
+	rmsummary_delete(l);
+	return set;
+}
+static int check_task_fit_in_connected_workers(struct work_queue *q, struct work_queue_task *t)
+{
+	char *key;
+	struct work_queue_worker *w;
+	hash_table_firstkey(q->worker_table);
+	
+	int bit_set = 0;
+	while(hash_table_nextkey(q->worker_table, &key, (void**)&w))
+	{
+		int new_set = check_task_fit_in_worker(q, t, w);
+		if (new_set == 0){
+			// Task could run on a currently connected worker, immediately return	
+			return 0;
+		}
+
+		// Inherit the unfit criteria for this task
+		bit_set = bit_set | new_set;
```

Comment:
I still think this causes the smallest worker to cause spurious messages.

Reply:
I have to put a lot more thought into how I would be able to solve this problem. for now, I will only increment on when a task does not fit to a resource, regardless if it does fit somewhere else. clearly using a bit set does not solve the nuances of potential workers that can fit, so i'll have to go back to the drawing board